system/Database/Sqlsrv/Builder.php

```diff
@@ -0,0 +1,691 @@
+	 * auto-enable
+	 *
+	 * @param string $table         The table name
+	 * @param array  $keys          The insert keys
+	 * @param array  $unescapedKeys The insert values
+	 *
+	 * @return string
+	 */
+	protected function _insert(string $table, array $keys, array $unescapedKeys): string
+	{
+		$fullTableName = $this->getFullName($table);
+
+		// check for index key
+		// TODO: implement check for this instad static $insertKeyPermission
+		// insert statement
+		$statement = 'INSERT INTO ' . $fullTableName . ' (' . implode(',', $keys) . ') VALUES (' . implode(', ', $unescapedKeys) . ')';
+
+		return $this->keyPermission ? $this->addIdentity($fullTableName, $statement) : $statement;
+	}
+
+	/**
+	 * Update statement
+	 *
+	 * Generates a platform-specific update string from the supplied data
+	 *
+	 * @param string $table  the Table name
+	 * @param array  $values the Update data
+	 *
+	 * @return string
+	 */
+	protected function _update(string $table, array $values): string
+	{
+		$valstr = [];
+
+		foreach ($values as $key => $val)
+		{
+			$valstr[] = $key . ' = ' . $val;
+		}
+
+		$statement = 'UPDATE ' . (empty($this->QBLimit) ? '' : 'TOP(' . $this->QBLimit . ') ') . $table . ' SET '
+				. implode(', ', $valstr) . $this->compileWhereHaving('QBWhere') . $this->compileOrderBy();
+
+		return $this->keyPermission ? $this->addIdentity($this->getFullName($table), $statement) : $statement;
+	}
+
+	/**
+	 * Increments a numeric column by the specified value.
+	 *
+	 * @param string  $column
+	 * @param integer $value
+	 *
+	 * @return boolean
+	 */
+	public function increment(string $column, int $value = 1)
+	{
+		$column = $this->db->protectIdentifiers($column);
+
+		if ($this->castTextToInt)
+		{
+			$values = [$column => "CONVERT(VARCHAR(MAX),CONVERT(INT,CONVERT(VARCHAR(MAX), {$column})) + {$value})"];
+		}
+		else
+		{
+			$values = [$column => "{$column} + {$value}"];
+		}
+		$sql = $this->_update($this->QBFrom[0], $values);
+
+		return $this->db->query($sql, $this->binds, false);
+	}
+
+	/**
+	 * Decrements a numeric column by the specified value.
+	 *
+	 * @param string  $column
+	 * @param integer $value
+	 *
+	 * @return boolean
+	 */
+	public function decrement(string $column, int $value = 1)
+	{
+		$column = $this->db->protectIdentifiers($column);
+
+		if ($this->castTextToInt)
+		{
+			$values = [$column => "CONVERT(VARCHAR(MAX),CONVERT(INT,CONVERT(VARCHAR(MAX), {$column})) - {$value})"];
+		}
+		else
+		{
+			$values = [$column => "{$column} + {$value}"];
+		}
+		$sql = $this->_update($this->QBFrom[0], $values);
+
+		return $this->db->query($sql, $this->binds, false);
+	}
+
+	/**
+	 * Get funn name of the table
+	 *
+	 * @param string $table
+	 *
+	 * @return string
+	 */
+	private function getFullName(string $table): string
+	{
+		if ($this->db->escapeChar === '"')
+		{
+			return '"' . $this->db->getDatabase() . '"."' . $this->db->schema . '"."' . str_replace('"', '', $table) . '"';
+		}
+		else
```

Comment:
no need else for return early, the check can be handled by rector at https://github.com/codeigniter4/CodeIgniter4/pull/3715

Reply:
I was removing these things from the code but the new rule for Rector will be very useful.